File: examples/anthropic-managed-agents/python/anthropic_managed_agents_e2b/session.py
from __future__ import annotations

import anthropic


def is_end_turn(event) -> bool:
    if getattr(event, "type", None) != "session.status_idle":
        return False
    return getattr(getattr(event, "stop_reason", None), "type", None) == "end_turn"


def stream_message(
    *,
    api_key: str,
    agent_id: str,
    environment_id: str,
    message: str,
):
    client = anthropic.Anthropic(api_key=api_key)
    session = client.beta.sessions.create(
        agent=agent_id,
        environment_id=environment_id,
    )
    print(f"session={session.id}", flush=True)

    with client.beta.sessions.events.stream(session.id) as stream:
        client.beta.sessions.events.send(
            session.id,
            events=[
                {
                    "type": "user.message",
                    "content": [{"type": "text", "text": message}],
                }
            ],
        )
        for event in stream:
            yield event
            if is_end_turn(event):
                break

File: examples/anthropic-managed-agents/javascript/src/session.ts
import Anthropic from "@anthropic-ai/sdk";

function isEndTurn(event: { type?: string; stop_reason?: { type?: string } | null }) {
  return event.type === "session.status_idle" && event.stop_reason?.type === "end_turn";
}

export async function streamMessage({
  apiKey,
  agentId,
  environmentId,
  message,
}: {
  apiKey: string;
  agentId: string;
  environmentId: string;
  message: string;
}) {
  const client = new Anthropic({ apiKey });
  const session = await client.beta.sessions.create({
    agent: agentId,
    environment_id: environmentId,
  });
  console.log(`session=${session.id}`);

  const stream = await client.beta.sessions.events.stream(session.id);
  await client.beta.sessions.events.send(session.id, {
    events: [
      {
        type: "user.message",
        content: [{ type: "text", text: message }],
      },
    ],
  });

  for await (const event of stream) {
    console.log(JSON.stringify(event));
    if (isEndTurn(event)) {
      break;
    }
  }
}
